### PR TITLE
Pilot: allow nested json access log format

### PR DESCRIPTION
**Pull request by max0ne**
Please provide a description for what this PR is for.

And to help us figure out who should review this PR, please 
put an X in all the areas that this PR affects.

[ ] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[X] Policies and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Fixes https://github.com/istio/istio/issues/28597

Please check any characteristics that apply to this pull request. 

[ ] Does not have any changes that may affect Istio users.

### CODE REVIEW

**Comment by max0ne** [comment]
@googlebot I signed it!

File: pilot/pkg/networking/core/v1alpha3/accesslog.go [hunk 1]
```diff
@@ -174,28 +174,20 @@ func buildFileAccessLogHelper(mesh *meshconfig.MeshConfig) *accesslog.AccessLog
...
-			} else {
+			if err := protomarshal.ApplyJSON(mesh.AccessLogFormat, &jsonLogValue); err != nil {
 				log.Errorf("error parsing provided json log format, default log format will be used: %v", err)
 			}
 		}
-		if jsonLog == nil {
-			jsonLog = EnvoyJSONLogFormat
+		jsonLogStruct := jsonLogValue.GetStructValue()
+		if jsonLogStruct == nil {
+			jsonLogStruct = EnvoyJSONLogFormat
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/accesslog.go:185]: should we log on this error like above?

File: pilot/pkg/networking/core/v1alpha3/accesslog.go [hunk 2]
```diff
@@ -174,28 +174,20 @@ func buildFileAccessLogHelper(mesh *meshconfig.MeshConfig) *accesslog.AccessLog
 			},
 		}
 	case meshconfig.MeshConfig_JSON:
-		var jsonLog *structpb.Struct
+		jsonLogValue := structpb.Value{}
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/accesslog.go:177]: nit: why change this from struct to value? Seems like we later skip anyways if its not a struct but then don't get a useful error message potentionally?

**Review by howardjohn — COMMENTED** [review]
Code LGTM. Will leave to @istio/wg-policies-and-telemetry-maintainers to ensure this makes sense

File: pilot/pkg/networking/core/v1alpha3/accesslog.go (hunk 2 above)
Comment (max0ne) [reply pilot/pkg/networking/core/v1alpha3/accesslog.go]: yeah good point, 8e163b2 changed it to just use one `structpb.Struct` to parse it

**Comment by howardjohn** [comment]
/ok-to-test

fyi @max0ne I am not an expert on access log so waiting on @istio/wg-policies-and-telemetry-maintainers to weigh in

**Comment by max0ne** [comment]
Hey, can anyone from @istio/wg-policies-and-telemetry-maintainers help take a look at this one?

**Review by mandarjog — APPROVED** [review]
Thanks for adding this.
Can you add doc to show “how to configure it for stackdriver?” 
@gargnupur can you review and suggest a place for the doc to go?

**Comment by gargnupur** [comment]
> Thanks for adding this.
> Can you add doc to show “how to configure it for stackdriver?”
> @gargnupur can you review and suggest a place for the doc to go?
This is cool!
We can add it here : https://istio.io/latest/docs/tasks/observability/logs/access-log/ -> add a section on “how to configure it for stackdriver?”

**Review by howardjohn — APPROVED** [review]

**Pull request merged**